### ISSUE TITLE
Won't activate with subscriptions 2.1-bleeding

### ISSUE THREAD

**Issue by rogercoathup** [Prospress/woocommerce-subscriptions-gifting#178]
I can't activate the gifting plugin when working with the 2.1-bleeding branch of subscriptions.

Receive the following message:

WooCommerce Subscriptions Gifting is inactive. WooCommerce Subscriptions Gifting requires the WooCommerce Subscriptions plugin to be active to work correctly. Please install & activate WooCommerce Subscriptions »

Is that an issue in the version check in gifting?

**Comment by rogercoathup** [comment]
The woocommerce_subscriptions_active_version option is: 2.1.0-beta-1

**Comment by james-allan** [comment]
Hi @rogercoathup 

The version string `2.1.0-beta-1` shouldn't be a problem. PHP's `version_compare()` can handle comparing version numbers which contain beta versions. 

I've just tried to replicate this locally. The `woocommerce_subscriptions_active_version` option on my local is `2.1.0-beta-1` and Gifting can be activated and is enabled just fine. 

Could you possibly provide us with more information to help replicate this?

**Comment by james-allan** [comment]
Ah. If you're receiving that specific error notice then it's not related to the Subscription version, it means it doesn't think Subscriptions is active at all.

The invalid version error would have looked like: 

![screen shot 2016-10-27 at 3 52 26 pm](https://cloud.githubusercontent.com/assets/8490476/19756089/896d2232-9c5d-11e6-9264-b64cfb2d882b.png)

----

Gifting uses `is_plugin_active( 'woocommerce-subscriptions/woocommerce-subscriptions.php' )` t check if Subscriptions is active, however if you've downloaded and installed the beta from https://github.com/Prospress/woocommerce-subscriptions/issues/1679, that plugin directory is different. 

ie:
`woocommerce-subscriptions-2.1.0-beta-1/woocommerce-subscriptions.php`
vs
 `woocommerce-subscriptions/woocommerce-subscriptions.php`

Installing other pre-released versions of Memberships and Subscriptions will likely cause the same issue so we'll need to come up with a solution for that. Thanks reporting it 😄

**Comment by rogercoathup** [comment]
@james-allan -- thanks, I must have read through the gifting code too quickly; I missed the is_plugin_active line.  I'll move subscriptions to the expected directory as a temporary workaround.

**Comment by rogercoathup** [comment]
@james-allan -- not sure if you want this on a separate thread:  I'm also seeing an issue if you rename / delete the subscriptions directory whilst gifting is active:

Fatal error: Class 'WCS_Query' not found in /vagrant/content/plugins/woocommerce-subscriptions-gifting/includes/class-wcsg-query.php on line 2

Gifting fallbacks gracefully if you deactivate subscriptions from wp-admin, but not if the directory is removed / renamed.

**Comment by james-allan** [comment]
> not sure if you want this on a separate thread: I'm also seeing an issue if you rename / delete the subscriptions directory whilst gifting is active

Nah that's fine to since they are linked to the same issue and  should probably be addressed with the same fix. I'm assuming what is happening here, is the `is_active_plugin()` check is passing but the plugin isn't actually active (since you've renamed/deleted it manually). I've seen other plugins have checks like:

```
is_plugin_active( 'woocommerce-subscriptions...' ) && class_exists( 'WC_Subscriptions' );
```

which should catch those cases.

**Comment by jorgeatorres** [comment]
This was addressed in #249.
